qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -477,6 +499,11 @@ def main(args=None):
         parser.error('--watch option must be used with --all')
     if args.watch and args.notify_monitor_layout:
         parser.error('--watch cannot be used with --notify-monitor-layout')
+    if not args.set_keyboard_layout:
+        args.set_keyboard_layout = args.watch
+    if args.set_keyboard_layout or os.path.exists('/etc/qubes-release'):
```

Comment:
Why exactly being in dom0 also triggers setting keyboard layout? After all, if GUI is running in dom0, there will be `qvm-start-daemon --watch --all`, so it should be enough, no?

Reply:
In fact, it misses a condition on the guivm-gui-agent as this will also apply in case of audiovm daemon.